Modules/Numerics/Optimizersv4/test/itkRegularStepGradientDescentOptimizerv4Test.cxx

```diff
@@ -384,6 +384,50 @@ itkRegularStepGradientDescentOptimizerv4Test(int, char *[])
                                                                                 currentLearningRateRelaxation);
   }
 
+  // Verify that the optimizer stays at the initial position if
+  // number of iterations is set to one.
+  std::cout << "\nCheck the optimizer position (parameters) are equal to the initial "
+               "value when number of iterations is set to one:"
+            << std::endl;
+  {
+    RSGv4TestMetric::Pointer metric = RSGv4TestMetric::New();
+    itkOptimizer->SetMetric(metric);
+    using ParametersType = RSGv4TestMetric::ParametersType;
+    const unsigned int spaceDimension = metric->GetNumberOfParameters();
+    ParametersType     initialPosition(spaceDimension);
+    initialPosition[0] = 100;
+    initialPosition[1] = -100;
+    metric->SetParameters(initialPosition);
+
+    itkOptimizer->SetNumberOfIterations(1);
+    try
+    {
+      itkOptimizer->StartOptimization();
+    }
+    catch (const itk::ExceptionObject & e)
+    {
+      return EXIT_FAILURE;
+    }
```

Comment:
The above block can be shortened using `ITK_TRY_EXPECT_NO_EXCEPTION(itkOptimizer->StartOptimization());`. It is preferred for the sake of compactness and to avoid boilerplate code.